File: zeus_1/zeus/zeus/static/js/resource.js
function valid_form(memoryform){
	with(memoryform)
	{
		if(brand.value=="")
		{
			alert("内存厂商不能为空");
			brand.focus();
			return false;
		}
		else if((quantity.value=="")||isNaN(quantity.value))
		{
			alert("内存数量不能为空且是数字");
			quantity.focus();
			return false;
		}
        else if(category.value=="")
        {
            alert("内存类型不能为空");
            category.focus();
            return false;
        }
  	    else if(capacity.value=="")
        {
            alert("内存容量不能为空");
            capacity.focus();
            return false;
        }
	    else if(frequency.value=="")
        {
            alert("内存频率不能为空");
            frequency.focus();
            return false;
        }
    	else if(check.value=="")
        {
            alert("内存校验不能为空");
            check.focus();
            return false;
        }
	    else if(maintainer.value=="")
        {
            alert("维护人不能为空");
            maintainer.focus();
            return false;
        }
   	    else if((available.value=="")||isNaN(available.value))
        {
            alert("内存可用不能为空且为数字");
            available.focus();
            return false;
        }

	}


}

File: zeus_1/zeus/zeus/views/resource.py
from django.shortcuts import render_to_response,HttpResponse
from zeus.models import Memory
import re
import simplejson

def resource(request):
	if request.POST:
		if request.POST['btnSubmit']=='add':
			memory = Memory(brand=request.POST['brand'], quantity=request.POST['quantity'],category=request.POST['category'],capacity=request.POST['capacity'],frequency=request.POST['frequency'],check=request.POST['check'],maintainer=request.POST['maintainer'],available=request.POST['available'])
       			memory.save()
		elif request.POST['btnSubmit']=='update':
			if request.POST['used_recorder']!='':
				pattern = re.compile(r'\*\d+')
				memory_available = int(request.POST['quantity'])
				for m in pattern.finditer(request.POST['used_recorder']):
					memory_used=m.group().replace('*','')
					memory_available=memory_available-int(memory_used)
			else:
				memory_available = request.POST['available']
			memory = Memory(id=request.POST['id'],brand=request.POST['brand'], quantity=request.POST['quantity'],category=request.POST['category'],capacity=request.POST['capacity'],frequency=request.POST['frequency'],check=request.POST['check'],maintainer=request.POST['maintainer'],available=memory_available,used_recorder=request.POST['used_recorder'])
			memory.save()
		elif request.POST['btnSubmit']=='delete':
			memory = Memory.objects.get(id=request.POST['id'])
			memory.delete()
	memorys=Memory.objects.all()
	return render_to_response('resource.html', {'memorys' : memorys})	

def ajax(request):
    data = {}
    #req = simplejson.loads(request.raw_post_data)
    value = request.POST.get("hello")

    #print type(req)
    #dict['name'] = 'hello'
    #print type(dict)
    data["lit"] = value
    json=simplejson.dumps(data)
    #print type(json)
    #return render_to_response('resource.html', {'dict' : dict})
    return HttpResponse(json)
